[Internal] Some runner tests fail on macOS

**Steps to reproduce:**

```shell
cd runner
go test -short -race $(go list ./... | grep -v /vendor/)
```

**Actual behaviour:**

```shell
?       github.com/dstackai/dstack/runner/cmd/runner    [no test files]
?       github.com/dstackai/dstack/runner/cmd/shim      [no test files]
?       github.com/dstackai/dstack/runner/consts        [no test files]
?       github.com/dstackai/dstack/runner/internal/api  [no test files]
?       github.com/dstackai/dstack/runner/internal/gerrors      [no test files]
?       github.com/dstackai/dstack/runner/internal/log  [no test files]
?       github.com/dstackai/dstack/runner/internal/runner/api   [no test files]
?       github.com/dstackai/dstack/runner/internal/schemas      [no test files]
ok      github.com/dstackai/dstack/runner/internal/common       1.208s
?       github.com/dstackai/dstack/runner/internal/shim/backends        [no test files]
?       github.com/dstackai/dstack/runner/internal/types        [no test files]
ok      github.com/dstackai/dstack/runner/internal/connections  1.410s
--- FAIL: TestExecutor_WorkingDir_Current (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10358ab60]

goroutine 12 [running]:
testing.tRunner.func1.2({0x1036c8040, 0x103a59ea0})
        /usr/local/go/src/testing/testing.go:1632 +0x2c4
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1635 +0x47c
panic({0x1036c8040?, 0x103a59ea0?})
        /usr/local/go/src/runtime/panic.go:791 +0x124
github.com/dstackai/dstack/runner/internal/executor.(*RunExecutor).SetJob(...)
        /Users/cheptsov/dstack/runner/internal/executor/executor.go:225
github.com/dstackai/dstack/runner/internal/executor.makeTestExecutor(0xc0003149c0)
        /Users/cheptsov/dstack/runner/internal/executor/executor_test.go:196 +0x480
github.com/dstackai/dstack/runner/internal/executor.TestExecutor_WorkingDir_Current(0xc0003149c0)
        /Users/cheptsov/dstack/runner/internal/executor/executor_test.go:24 +0x74
testing.tRunner(0xc0003149c0, 0x1037534a8)
        /usr/local/go/src/testing/testing.go:1690 +0x188
created by testing.(*T).Run in goroutine 1
        /usr/local/go/src/testing/testing.go:1743 +0x5e4
FAIL    github.com/dstackai/dstack/runner/internal/executor     0.240s
--- FAIL: TestGetAMDGPUMetrics_OK (0.00s)
    metrics_test.go:12: 
                Error Trace:    /Users/cheptsov/dstack/runner/internal/metrics/metrics_test.go:12
                Error:          Received unexpected error:
                                could not read /proc/self/mountinfo: open /proc/self/mountinfo: no such file or directory
                Test:           TestGetAMDGPUMetrics_OK
--- FAIL: TestGetAMDGPUMetrics_ErrorGPUUtilNA (0.00s)
    metrics_test.go:43: 
                Error Trace:    /Users/cheptsov/dstack/runner/internal/metrics/metrics_test.go:43
                Error:          Received unexpected error:
                                could not read /proc/self/mountinfo: open /proc/self/mountinfo: no such file or directory
                Test:           TestGetAMDGPUMetrics_ErrorGPUUtilNA
FAIL
FAIL    github.com/dstackai/dstack/runner/internal/metrics      0.402s
ok      github.com/dstackai/dstack/runner/internal/repo 1.331s
ok      github.com/dstackai/dstack/runner/internal/shim 1.593s
ok      github.com/dstackai/dstack/runner/internal/shim/api     1.232s
ok      github.com/dstackai/dstack/runner/internal/shim/dcgm    1.752s
ok      github.com/dstackai/dstack/runner/internal/shim/host    1.412s
FAIL
```

**System info:**

- macOS Sequoia 15.4.1

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -123,7 +123,10 @@ jobs:
     defaults:
       run:
         working-directory: runner
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
     steps:
       - uses: actions/checkout@v4
       - name: Set up Go

--- a/runner/internal/executor/executor.go
+++ b/runner/internal/executor/executor.go
@@ -28,6 +28,12 @@ import (
 	"github.com/prometheus/procfs"
 )
 
+type ConnectionTracker interface {
+	GetNoConnectionsSecs() int64
+	Track(ticker <-chan time.Time)
+	Stop()
+}
+
 type RunExecutor struct {
 	tempDir    string
 	homeDir    string
@@ -52,11 +58,7 @@ type RunExecutor struct {
 	timestamp       *MonotonicTimestamp
 
 	killDelay         time.Duration
-	connectionTracker interface {
-		GetNoConnectionsSecs() int64
-		Track(ticker <-chan time.Time)
-		Stop()
-	}
+	connectionTracker ConnectionTracker
 }
 
 // stubConnectionTracker is a no-op implementation for when procfs is not available (only required for tests on darwin)
@@ -79,11 +81,7 @@ func NewRunExecutor(tempDir string, homeDir string, workingDir string, sshPort i
 	}
 
 	// Try to initialize procfs, but don't fail if it's not available (e.g., on macOS)
-	var connectionTracker interface {
-		GetNoConnectionsSecs() int64
-		Track(ticker <-chan time.Time)
-		Stop()
-	}
+	var connectionTracker ConnectionTracker
 
 	if runtime.GOOS == "linux" {
 		proc, err := procfs.NewDefaultFS()

--- a/runner/internal/executor/executor.go
+++ b/runner/internal/executor/executor.go
@@ -11,6 +11,7 @@ import (
 	"os/exec"
 	osuser "os/user"
 	"path/filepath"
+	"runtime"
 	"strconv"
 	"strings"
 	"sync"
@@ -51,8 +52,19 @@ type RunExecutor struct {
 	timestamp       *MonotonicTimestamp
 
 	killDelay         time.Duration
-	connectionTracker *connections.ConnectionTracker
+	connectionTracker interface {
+		GetNoConnectionsSecs() int64
+		Track(ticker <-chan time.Time)
+		Stop()
+	}
 }
+
+// stubConnectionTracker is a no-op implementation for when procfs is not available (only required for tests on darwin)
+type stubConnectionTracker struct{}
+
+func (s *stubConnectionTracker) GetNoConnectionsSecs() int64   { return 0 }
+func (s *stubConnectionTracker) Track(ticker <-chan time.Time) {}
+func (s *stubConnectionTracker) Stop()                         {}
 
 func NewRunExecutor(tempDir string, homeDir string, workingDir string, sshPort int) (*RunExecutor, error) {
 	mu := &sync.RWMutex{}
@@ -65,15 +77,29 @@ func NewRunExecutor(tempDir string, homeDir string, workingDir string, sshPort i
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse current user uid: %w", err)
 	}
-	proc, err := procfs.NewDefaultFS()
-	if err != nil {
-		return nil, fmt.Errorf("failed to initialize procfs: %w", err)
+
+	// Try to initialize procfs, but don't fail if it's not available (e.g., on macOS)
+	var connectionTracker interface {
+		GetNoConnectionsSecs() int64
+		Track(ticker <-chan time.Time)
+		Stop()
 	}
-	connectionTracker := connections.NewConnectionTracker(connections.ConnectionTrackerConfig{
-		Port:            uint64(sshPort),
-		MinConnDuration: 10 * time.Second, // shorter connections are likely from dstack-server
-		Procfs:          proc,
-	})
+
+	if runtime.GOOS == "linux" {
+		proc, err := procfs.NewDefaultFS()
+		if err != nil {
+			return nil, fmt.Errorf("failed to initialize procfs: %w", err)
+		}
+		connectionTracker = connections.NewConnectionTracker(connections.ConnectionTrackerConfig{
+			Port:            uint64(sshPort),
+			MinConnDuration: 10 * time.Second, // shorter connections are likely from dstack-server
+			Procfs:          proc,
+		})
+	} else {
+		// Use stub connection tracker (only required for tests on darwin)
+		connectionTracker = &stubConnectionTracker{}
+	}
+
 	return &RunExecutor{
 		tempDir:    tempDir,
 		homeDir:    homeDir,

--- a/runner/internal/metrics/metrics_test.go
+++ b/runner/internal/metrics/metrics_test.go
@@ -1,6 +1,7 @@
 package metrics
 
 import (
+	"runtime"
 	"testing"
 
 	"github.com/dstackai/dstack/runner/internal/schemas"
@@ -8,6 +9,9 @@ import (
 )
 
 func TestGetAMDGPUMetrics_OK(t *testing.T) {
+	if runtime.GOOS == "darwin" {
+		t.Skip("Skipping on macOS")
+	}
 	collector, err := NewMetricsCollector()
 	assert.NoError(t, err)
 
@@ -39,6 +43,9 @@ func TestGetAMDGPUMetrics_OK(t *testing.T) {
 }
 
 func TestGetAMDGPUMetrics_ErrorGPUUtilNA(t *testing.T) {
+	if runtime.GOOS == "darwin" {
+		t.Skip("Skipping on macOS")
+	}
 	collector, err := NewMetricsCollector()
 	assert.NoError(t, err)
 	metrics, err := collector.getAMDGPUMetrics("gpu,gfx,gfx_clock,vram_used,vram_total\n0,N/A,N/A,283,196300\n")

